js/background.js

```diff
@@ -277,4 +280,38 @@
             timestamp: timestamp, source: pushMessage.source
         });
     }
+
+    var App = Backbone.Model.extend({
+        initialize: function(opts){
+            this.inboxView = null;
+        },
+        getAppView: function(destWindow){
+
+            var self = this;
+
+            return ConversationController.updateInbox().then(function() {
+                try {
+                    if (self.inboxView) { self.inboxView.remove(); }
+                    self.inboxView = new Whisper.InboxView({model: self, window: destWindow});
+                    window.openConversation = function(conversation) {
```

Comment:
Thanks! I added a patch that makes sure the function is defined on the destination window.

So this change is a consequence of moving the UI initialization machinery over to the background window.  I see two benefits to doing that. First is that we're grouping all our classes, and functions in the same global scope. So no need to juggle between windows to update view state based on the socket status. The second (hypothetical) is that it may make it easier to insert the signal desktop UI into a tab. Which could be nice if porting to chrome extension.